cli/src/main/kotlin/uk/gov/justice/digital/client/BlockingDomainClient.kt

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.digital.client
+
+import io.micronaut.core.type.Argument
+import io.micronaut.http.HttpHeaders.*
+import io.micronaut.http.HttpRequest.GET
+import io.micronaut.http.MediaType.APPLICATION_JSON
+import io.micronaut.http.client.HttpClient
+import io.micronaut.http.client.annotation.Client
+import io.micronaut.http.uri.UriBuilder
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+import uk.gov.justice.digital.model.Domain
+
+interface DomainClient {
+    fun getDomains(): Array<Domain>
+    fun getDomainWithName(name: String): Domain?
+}
+
+/**
+ * Blocking client, suitable for CLI use.
+ */
+@Singleton
+class BlockingDomainClient : DomainClient {
+
+    @field:Client("\${micronaut.http.services.domain.url}")
+    @Inject
+    private lateinit var client: HttpClient
+
+    override fun getDomains(): Array<Domain> = client.get<Array<Domain>>(DOMAIN_RESOURCE)
+
+    override fun getDomainWithName(name: String): Domain? {
```

Comment:
can we rename the method to getDomainByName()?

Reply:
This name is consistent with naming conventions used elsewhere in the project so I'll leave as is. These methods will likely change anyway as the search functionality is built out.